lib/fluent/plugin/out_influxdb.rb

```diff
@@ -1,53 +1,58 @@
-               :desc => "The password of the user."
-  config_param :retry, :integer, :default => nil,
-               :desc => 'The finite number of retry times. default is infinite'
-  config_param :time_key, :string, :default => 'time',
-               :desc => 'Use value of this tag if it exists in event instead of event timestamp'
-  config_param :time_precision, :string, :default => 's',
-               :desc => <<-DESC
+  config_param :user, :string,  default: 'root',
+               desc: "The DB user of influxDB, should be created manually."
+  config_param :password, :string,  default: 'root', secret: true,
+               desc: "The password of the user."
+  config_param :retry, :integer, default: nil,
+               desc: 'The finite number of retry times. default is infinite'
+  config_param :time_key, :string, default: 'time',
+               desc: 'Use value of this tag if it exists in event instead of event timestamp'
+  config_param :time_precision, :string, default: 's',
+               desc: <<-DESC
 The time precision of timestamp.
 You should specify either hour (h), minutes (m), second (s),
 millisecond (ms), microsecond (u), or nanosecond (n).
 DESC
-  config_param :use_ssl, :bool, :default => false,
-               :desc => "Use SSL when connecting to influxDB."
-  config_param :verify_ssl, :bool, :default => true,
-               :desc => "Enable/Disable SSL Certs verification when connecting to influxDB via SSL."
-  config_param :tag_keys, :array, :default => [],
-               :desc => "The names of the keys to use as influxDB tags."
-  config_param :sequence_tag, :string, :default => nil,
-               :desc => <<-DESC
+  config_param :use_ssl, :bool, default: false,
+               desc: "Use SSL when connecting to influxDB."
+  config_param :verify_ssl, :bool, default: true,
+               desc: "Enable/Disable SSL Certs verification when connecting to influxDB via SSL."
+  config_param :tag_keys, :array, default: [],
+               desc: "The names of the keys to use as influxDB tags."
+  config_param :sequence_tag, :string, default: nil,
+               desc: <<-DESC
 The name of the tag whose value is incremented for the consecutive simultaneous
 events and reset to zero for a new event with the different timestamp.
 DESC
-  config_param :retention_policy_key, :string, :default => nil,
-               :desc => "The key of the key in the record that stores the retention policy name"
-  config_param :default_retention_policy, :string, :default => nil,
-               :desc => "The name of the default retention policy"
+  config_param :retention_policy_key, :string, default: nil,
+               desc: "The key of the key in the record that stores the retention policy name"
+  config_param :default_retention_policy, :string, default: nil,
+               desc: "The name of the default retention policy"
 
+  config_section :buffer do
+    config_set_default :@type, DEFAULT_BUFFER_TYPE
+  end
```

Comment:
If this plugin prefer to use `tag` as default chunk key, specify `config_set_default :chunk_keys, ['tag']` in `config_section :buffer`.